### ISSUE TITLE
[planning] Retrospective and planning for 1905-1 (04/25/2019)

### ISSUE THREAD

**Issue by molant** [webhintio/hint#2304]
**Time**: 04/25/2019 11:00 PST
**Meeting link (participants)**: https://zoom.us/j/172034153
**YouTube recording**: https://www.youtube.com/watch?v=aBxgqg9d7aE
**Agenda items**:

* [x] What tasks are moving to the next sprint
* [x] What we've finished (or almost finished for the sprint)
* [x] Status of the browser extension
* [x] Backlog for next sprint: main goal is to have the extension ready to ship on any store with an integrated design (🤞)
  * [x] Are there still WebExtension API blockers?
  * [x] Any requirements? Footprint? a11y? i18n?
  * [x] Character offset when linking to sources section
  * [x] Link to network panel
  * [x] Collecting input for trying out the extension with Firefox users.
  * [x] Go-to-market + promotion plan for WebHint in general.
  * [x] Interested in JS compat

--------------------------------------------------

# Meeting notes

## General

* Bit of miss this sprint. Lots of internal events and meetings delayed a few of the tasks and some issues we weren't expected or are taking longer than expected, notably: 
  * creating a new staging environment for the online service
  * puppeteer browser lifecycle
  * triming compat data and the compat api issue
* Moving all the pending tasks to the next sprint and not adding nothing new for now (bugs as they come along)
* **Find better time for people in Europe**: @molant to create poll (or similar) to find a better time for everyone for next meeting
* Main goals for 1905-1 are:
  * `connector-puppeteer`
  * Get the extension in an almost shippable state in Edge and Firefox (only feedback from users and/or minor issues)
  * Continue the migration to serverless

## Connector-chromium

* [ ] There's puppeteer for Firefox currently on the works, need to document the methods we are using to see if it will work once it's out of beta (@molant)
* [ ] Rename `connector-chromium` to `connector-puppeteer` (@molant)

## Extension

* [x] Link to the elements pannel with an offset to point to the right place and link to the network pannel (latter requires a new API). @digitarald to update the bugs in Mozilla's side. → **https://bugzilla.mozilla.org/show_bug.cgi?id=1548467**
* Extension uses an event page which is not supported by Firefox right now. Extension is still working but there's a memory penalty for people that have it installed and not using it in the current session
* Requirements for publishing:
  * Get a designed integrated. @violasong working on the design and will be sharing more mockups. Once #1670 (and maybe #1669) are done @molant will create a new issue to ping Harald so Mozilla can start implementing it.
  * We will also do some a11y checks on our side.

* Data collection
  * Can't hot load external libraries.
  * [ ] @antross to review [the guidance](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/AMO/Policy/Reviews#Data_Disclosure_Collection_and_Management) to see if it applies to our case (we bundle AppInsights directly and no code is downloaded during execution).
  * No enforced guidance to ask for permission (could be a one time pop up, ToU, etc.)

* Webhint promotion and user satisfaction:
  * No "artificial" milestone, when we are ready we ship. If extension is ready first on Firefox then we publish it. 
  * The idea is to use all webhint and Edge network to promote the project regardless of what's being published: Official twitter accounts, blogpost if needed, events where we participate and such.
  * We will start testing and sharing a bundle soon after #2287 gets merged to gather feedback and apply it before it's officially published to the store
  * For Firefox we need to [sign the extension](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/web-ext_command_reference#web-ext_sign) so it's easier to test it before publishing
  * Telemetry: Mozilla can check usage of the extension after installation
  * Surveys: There's infrastructure in Mozilla's side to get surveys to targeted users. @digitarald, @brendyna and @paulgildea to partner to craft the right questions.

## MDN Data

* JS Compat
  * Infrastructure already in place. We will start first with the static analysis and if there are scenarios that don't get covered switch to something different.
  * Extension and compat api with CSS have a higher priority right now so it will take a few sprints to get to this.

**Comment by violasong** [comment]
Re: the adaptation of Stephanie's mockups I'm working on - [here's how it looks](https://mozilla.invisionapp.com/share/V3RPCITNG2E#/359891044_webhint_01) so far.

<img width="902" alt="image" src="https://user-images.githubusercontent.com/55994/56698871-3c141a00-66a8-11e9-91ca-0964862b07d2.png">

**Comment by digitarald** [comment]
Related to the addons discussion: [Data collection in addon](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/AMO/Policy/Reviews#Data_Disclosure_Collection_and_Management)

**Comment by digitarald** [comment]
@molant about [signing extensions](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/web-ext_command_reference#web-ext_sign)

**Comment by molant** [comment]
Closing as the new sprint is approaching and I've done some cleaning #2403
